airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -112,10 +117,66 @@ def should_retry(self, response: requests.Response) -> bool:
         else:
             return super().should_retry(response)
 
+
+class ShopifyDeletedEventsStream(ShopifyStream):
+
+    data_field = "events"
+    primary_key = "id"
+    cursor_field = "deleted_at"
+
     @property
-    @abstractmethod
-    def data_field(self) -> str:
-        """The name of the field in the response which contains the data"""
+    def availability_strategy(self) -> None:
+        """
+        No need to apply the `availability strategy` for this service stream.
+        """
+        return None
+
+    def __init__(self, config: Dict, deleted_events_api_name: str):
+        self.deleted_events_api_name = deleted_events_api_name
+        super().__init__(config)
+
+    def path(self, **kwargs) -> str:
+        return f"{self.data_field}.json"
+
+    def get_json_schema(self) -> None:
+        """
+        No need to apply the `schema` for this service stream.
+        """
+        return {}
+
+    def produce_deleted_records_from_events(self, delete_events: Iterable[Mapping[str, Any]] = []) -> None:
+        for event in delete_events:
+            yield {
+                "id": event["subject_id"],
+                self.cursor_field: event["created_at"],
+                "updated_at": event["created_at"],
+                "message": event["message"],
```

Comment:
I recommend changing this to `deleted_*` e.g: `deleted_message`, `deleted_description`, etc...